.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,79 @@
+# reference:
+#   - https://github.com/actions/cache
+#   - https://github.com/actions/checkout
+
+name: benchmark-check
+
+on:
+  pull_request:
+
+  push:
+    branches:
+      - "main"
+      - "v*x"
+      - "!auto-update-lockfiles"
+      - "!pre-commit-ci-update-config"
+      - "!dependabot/*"
+    tags:
+      - "v*"
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  benchmark:
+    name: "performance benchmarks (py${{ matrix.python-version }})"
+
+    runs-on: ubuntu-latest
+
+    env:
+      PY_VER: "3.10"
+      IRIS_TEST_DATA_LOC_PATH: benchmarks
+      IRIS_TEST_DATA_PATH: benchmarks/iris-test-data
+      IRIS_TEST_DATA_VERSION: "2.19"
+      #: If you change the IRIS_SOURCE here you will also need to change it in
+      #: the noxfile and the tests and wheel workflows.
+      IRIS_SOURCE: "github:main"
+      ENV_CACHE_BUILD: "0"
+      TEST_DATA_CACHE_BUILD: "0"
+
+    strategy:
+      matrix:
+        python-version: ["3.10"]
+    
+    steps:
+      - name: "checkout"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - run: git checkout HEAD^2
```

Comment:
Worth a comment here, to explain that this checks out the `HEAD` of the Pull Request, instead of using GHA's default simulated merge commit.

Reply:
❗ This block should also be conditional - only run when on a pull request.

The same action also runs post-merge, and in that situation should run on the default commit with no modifications.

That's why the below line also has a similar check:

https://github.com/SciTools-incubator/iris-esmf-regrid/blob/e6c4cd769d821ce0b9a340df7b0bfffe10a06773/.github/workflows/benchmark.yml#L69

Reply:
see bb271a9